### PR TITLE
Subtract gas in eval dispatch.

### PR DESCRIPTION
**Pull request by happi**
Use some gas for each instruction.
Note: Not tuned in any way yet.

### CODE REVIEW

File: apps/aefate/src/aefa_fate_eval.erl [hunk 1]
```diff
@@ -5,345 +5,457 @@
 -module(aefa_fate_eval).
 -export([eval/2]).
 
-eval('RETURN', EngineState) ->
-    aefa_fate_op:return(EngineState);
+eval('RETURN',  #{ gas := Gas } = EngineState) ->
+    EngineState1 = EngineState#{ gas => Gas - 2},
```
Comment (ThomasArts) [inline apps/aefate/src/aefa_fate_eval.erl]: Would it not be nicer and shorter to write:
`aefa_fate_op:return( EngineState#{ gas => Gas - 2}).`
Saves a few lines here and there.

or a little wrapper decrease_gas(2, EngineState).
A matter of taste, of course. Advantage for such a wrapper is that you can check in  the wrapper that gas does not get negative... but now you check in return, which is fine as well.

Comment (happi) [reply apps/aefate/src/aefa_fate_eval.erl]: Fixed

**Comment by happi** [comment]
There are actually Fate tests that checks the gas now. Fix needed.

**Comment by lucafavatella** [comment]
A past failure in CI in this ticket was independent and [already tracked](https://www.pivotaltracker.com/story/show/165122907/comments/201657052).

**Pull request merged**